src/URI.js

```diff
@@ -1615,6 +1616,13 @@
       segments.unshift('');
     }
 
+    // remove leading and trailing slashes
+    // from segment
+    for (var i=0; i < segments.length; i++) {
```

Comment:
the problem is that we are setting this value in multiple places in the `if` statement in this method, and I didn't want to add more complexity to that part of the code.

I can change the code to do this replace in each branch if you think that's better.